drivers/gpio/gpio_smartbond.c

```diff
@@ -318,12 +418,16 @@ static const struct gpio_driver_api gpio_smartbond_drv_api_funcs = {
 		.wkup_regs = (volatile struct gpio_smartbond_wkup_regs *)		\
 						DT_INST_REG_ADDR_BY_NAME(id, wkup),	\
 		.wkup_trig_select = id,							\
+		GPIO_PM_DEVICE_CFG(ngpios, DT_PROP(DT_NODELABEL(gpio##id), ngpios))	\
```

Comment:
this is wrong, there's no guarantee instance '0' will be 'gpio0'.

Reply:
fixed, `gpio_smartbond_saved_state_<n>` generated for each gpio port if needed when **PM_DEVICE** is enabled